src/client/AM.PA.MonitoringTool/SlackTracker/Daemon.cs

```diff
@@ -0,0 +1,345 @@
+        }
+
+        public override bool IsEnabled()
+        {
+            return Database.GetInstance().GetSettingsBool(Settings.TRACKER_ENABLED_SETTING, Settings.IsEnabledByDefault);
+        }
+
+        public override bool IsFirstStart { get { _wasFirstStart = !Database.GetInstance().HasSetting(Settings.TRACKER_ENABLED_SETTING); return !Database.GetInstance().HasSetting(Settings.TRACKER_ENABLED_SETTING); } }
+
+
+        public void InternalStart()
+        {
+            try
+            {
+                CheckIfSecretsAreAvailable();
+                CheckIfTokenIsAvailable();
+
+                if (IsEnabled())
+                {
+                    Logger.WriteToConsole("Start Slack Tracker");
+                    CreateSlackPullTimer();
+                    IsRunning = true;
+                }
+            }
+            catch (Exception e)
+            {
+                Logger.WriteToLogFile(e);
+            }
+        }
+
+        public override void Start()
+        {
+            _isPApaused = false;
+            InternalStart();
+        }
+
+        public void InternalStop()
+        {
+            if (_slackTimer != null)
+            {
+                _slackTimer.Enabled = false;
+            }
+            IsRunning = false;
+        }
+
+        public override void Stop()
+        {
+            _isPApaused = true;
+            InternalStop();
+        }
+
+        public override void CreateDatabaseTablesIfNotExist()
+        {
+            DatabaseConnector.CreateSlackTables();
+        }
+
+        public override void UpdateDatabaseTables(int version)
+        {
+
+        }
+
+        //Called when new data should be pull from the slack API
+        private void OnPullFromSlack(object sender, ElapsedEventArgs eventArgs)
+        {
+            _slackTimer.Interval = Settings.SYNCHRONIZE_INTERVAL;
+
+            Logger.WriteToConsole("Try to sync with Slack");
+
+            try
+            {
+                DateTimeOffset latestSync;
+                
+                if (DatabaseConnector.GetLastTimeSynced() == "never")
+                {
+                    Logger.WriteToConsole("Sync for the First Time with slack");
+                    latestSync = DateTimeOffset.MinValue;
+                }
+                else
+                {
+                    latestSync = DateTimeOffset.Parse(DatabaseConnector.GetLastTimeSynced());
+                }
+
+                Logger.WriteToConsole("Latest sync date: " + latestSync.ToString(Settings.FORMAT_DAY_AND_TIME));
+                if(latestSync != DateTimeOffset.MinValue) {latestSync = latestSync.AddDays(-1);}
+
+
+                GetChannels(latestSync);
+                GetLogs(latestSync);
+                GetUsers(latestSync);
+                AnalyseLogs();
+
+                Database.GetInstance().SetSettings(Settings.LAST_SYNCED_DATE, DateTimeOffset.Now.ToString(Settings.FORMAT_DAY_AND_TIME));
```

Comment:
do you need the setting LAST_SYNCED_DATE as a setting or can it be just logged in the log-table?